src/modules/person/api/person.controller.ts

```diff
@@ -438,4 +449,55 @@ export class PersonController {
 
         return { ok: true, value: personResult.value.newPassword! };
     }
+
+    @Put(':personId/lock-user')
+    @HttpCode(HttpStatus.ACCEPTED)
+    @ApiOkResponse({ description: 'User has been successfully updated.', type: PersonLockResponse })
+    @ApiNotFoundResponse({ description: 'The person was not found.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to perform operation.' })
+    @ApiInternalServerErrorResponse({ description: 'An internal server error occurred.' })
+    public async lockPerson(
+        @Param('personId') personId: string,
+        @Body() lockUserBodyParams: LockUserBodyParams,
+        @Permissions() permissions: PersonPermissions,
+    ): Promise<PersonLockResponse> {
+        const personResult: Result<Person<true>> = await this.personRepository.getPersonIfAllowed(
+            personId,
+            permissions,
+        );
+
+        if (!personResult.ok) {
+            throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
+                SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(
+                    new MissingPermissionsError('Person not found or no permissions.'),
+                ),
+            );
+        }
+
+        if (!personResult.value?.keycloakUserId) {
+            throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
+                SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(new EntityNotFoundError('Person', personId)),
+            );
+        }
+
+        const tempAttributes: Record<string, string> = {};
+        tempAttributes[LOCK_KEYS[0] as string] = lockUserBodyParams.locked_from;
+        tempAttributes[LOCK_KEYS[1] as string] = new Date().toISOString();
+
+        const customAttributes: Record<string, string> = tempAttributes;
+
+        const result: Result<void, DomainError> = await this.keycloakUserService.updateKeycloakUserStatus(
+            personResult.value.keycloakUserId,
+            !lockUserBodyParams.lock,
+            customAttributes,
+        );
+        if (!result.ok) {
+            throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
+                SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(
+                    new EntityCouldNotBeUpdated('Person', personId),
+                ),
+            );
+        }
+        return new PersonLockResponse(`User has been successfully ${lockUserBodyParams.lock ? '' : 'un'}locked.`);
```

Comment:
This is a new endpoint and we don't need to adhere to schulconnex spec (right?).
Prefer throwing our domain errors directly and handle them in the specific filters (We already have `AuthenticationExceptionFilter` and `PersonExceptionFilter` in this controller, maybe expand those if needed)

Reply:
I refactored this to throw domain errors, two of which are new. @marode-cap could you please verify if I implemented this correctly?

Reply:
@clauyan looks okay from my perspective